factory/runType/sovereignRunTypeComponents.go

```diff
@@ -0,0 +1,115 @@
+package runType
+
+import (
+	"fmt"
+
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-go/dataRetriever/requestHandlers"
+	"github.com/multiversx/mx-chain-go/epochStart/bootstrap"
+	"github.com/multiversx/mx-chain-go/errors"
+	"github.com/multiversx/mx-chain-go/process/block"
+	"github.com/multiversx/mx-chain-go/process/block/preprocess"
+	"github.com/multiversx/mx-chain-go/process/coordinator"
+	"github.com/multiversx/mx-chain-go/process/peer"
+	"github.com/multiversx/mx-chain-go/process/smartContract/hooks"
+	"github.com/multiversx/mx-chain-go/process/sync"
+	"github.com/multiversx/mx-chain-go/process/sync/storageBootstrap"
+	"github.com/multiversx/mx-chain-go/process/track"
+)
+
+// SovereignRunTypeComponentsFactoryArgs holds the arguments needed for creating a state components factory
+type SovereignRunTypeComponentsFactoryArgs struct {
```

Comment:
I don't think we need this extra struct here.

We can have do it the same way we create other sovereign components: 

```go
func NewSovereignRunTypeComponentsFactory(args *runTypeComponentsFactory) (*sovereignRunTypeComponentsFactory, error)
```

Reply:
removed